Sources/ZcashLightClientKit/Synchronizer.swift

```diff
@@ -90,6 +90,23 @@ public protocol Synchronizer {
     /// Stop this synchronizer. Implementations should ensure that calling this method cancels all jobs that were created by this instance.
     func stop() throws
 
+    /// Call this when application did enter background. Synchronizer stops it's work.
+    ///
+    /// Synchronizer need some time to stop a work. So before calling this method use `UIApplication.shared.beginBackgroundTask` to request some
+    /// background execution time from operating system.
+    ///
+    /// - Parameters:
+    ///   - resumeSyncWhenAppGoesToForeground: Indicates if sync process should automatically start after app goes back to foreground. To support
+    ///                                        this call `applicationWillEnterForeground` when app will enter foreground.
+    ///   - backgroundWorkFinished: Closure is called when synchronizer finishes stop of the work.
+    func applicationDidEnterBackground(resumeSyncWhenAppGoesToForeground: Bool, backgroundWorkFinished: @escaping () -> Void)
```

Comment:
```suggestion
    func applicationDidEnterBackground(resumeOnForeground: Bool, synchronizerStopped: @escaping () -> Void)
```

I think these names are more concise. Also stopped doesn't imply that synchronizer finished syncing, it just refers to it stopping.

Reply:
Fixed